### PR TITLE
Huge refactorig for consistent col/row numbering

### PR DESCRIPTION
**Pull request by albenik**
# PR Details

Huge refactorig for consistent col/row numbering

## Description

Started from simply changing ToALphaString()/TitleToNumber() logic and related fixes.
But have to go deeper, do fixes, after do related fixes and again and again.

Major improvements:

1. Tests made stronger again (But still be weak).
2. "Empty" returns for incorrect input replaces with panic.
3. Check for correct col/row/cell naming & addressing by default.
4. Removed huge amount of duplicated code.
5. Removed ToALphaString(), TitleToNumber() and it helpers functions at all,
and replaced with SplitCellName(), JoinCellName(), ColumnNameToNumber(), ColumnNumberToName(), CellNameToCoordinates(), CoordinatesToCellName().
6. Minor fixes for internal variable naming for code readability (ex. col, row for input params, colIdx, rowIdx for slice indexes etc).
7. Excel time conversion bug fixed.

## Related Issue

https://github.com/360EntSecGroup-Skylar/excelize/issues/349

## Motivation and Context

Only consistent cell addressing was target. But I have a domino effect.

## How Has This Been Tested

New unit test written for any new functions.
Old test updated for edge cases.

All passed.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by albenik** [comment]
Fixed

File: rows.go [hunk 1]
```diff
@@ -329,12 +336,11 @@ func (f *File) GetRowVisible(sheet string, row int) bool {
 //    xlsx.SetRowOutlineLevel("Sheet1", 2, 1)
 //
 func (f *File) SetRowOutlineLevel(sheet string, row int, level uint8) {
-	xlsx := f.workSheetReader(sheet)
 	if row < 1 {
-		return
+		panic(newInvalidRowNumberError(row)) // Fail fats to avoid possible future side effects!
```
Comment (mlh758) [inline rows.go]: typo: `Fail fast` instead of `Fail fats`

Comment (xuri) [reply rows.go]: Thank you. I have fixed this typo.
